src/tools/agent-ctl/src/client/command.rs

```diff
@@ -1,520 +1,1710 @@
 
 impl AgentCmd for AddARPNeighbors {
     fn exec(
+        &self,
         ctx: &Context,
         client: &AgentServiceClient,
-        health: &HealthClient,
-        options: &mut Options,
+        _health: &HealthClient,
+        _options: &mut Options,
         args: &str,
-    ) -> Result<()> {
-        super::agent_cmd_sandbox_add_arp_neighbors(ctx, client, health, options, args)
+    ) -> (Result<()>, bool) {
+        let req: AddARPNeighborsRequest = match utils::make_request(args) {
+            Ok(res) => res,
+            Err(e) => {
+                return (
+                    Err(anyhow!("{:?}", e).context(REQUEST_BUILD_FAIL_MESSAGE)),
+                    false,
+                )
+            }
+        };
+
+        // FIXME: Implement fully.
+        eprintln!("FIXME: 'AddARPNeighbors' not fully implemented");
+
+        debug!(sl!(), "sending request"; "request" => format!("{:?}", req));
+
+        match client.add_arp_neighbors(ctx.clone(), &req) {
+            Ok(res) => {
+                info!(sl!(), "response received"; "response" => format!("{:?}", res));
+                (Ok(()), false)
+            }
+            Err(e) => (Err(anyhow!("{:?}", e).context(ERR_API_FAILED)), false),
+        }
     }
 }
 
 struct AddSwap;
 
 impl AgentCmd for AddSwap {
     fn exec(
+        &self,
         ctx: &Context,
         client: &AgentServiceClient,
-        health: &HealthClient,
-        options: &mut Options,
-        args: &str,
-    ) -> Result<()> {
-        super::agent_cmd_sandbox_add_swap(ctx, client, health, options, args)
+        _health: &HealthClient,
+        _options: &mut Options,
+        _args: &str,
+    ) -> (Result<()>, bool) {
+        let req = AddSwapRequest::default();
+
+        // FIXME: Implement 'AddSwap' fully.
+        eprintln!("FIXME: 'AddSwap' not fully implemented");
+
+        debug!(sl!(), "sending request"; "request" => format!("{:?}", req));
+
+        match client.add_swap(ctx.clone(), &req) {
+            Ok(res) => {
+                info!(sl!(), "response received"; "response" => format!("{:?}", res));
+                (Ok(()), false)
+            }
+            Err(e) => (Err(anyhow!("{:?}", e).context(ERR_API_FAILED)), false),
+        }
     }
 }
 
 struct Check;
 
 impl AgentCmd for Check {
     fn exec(
+        &self,
         ctx: &Context,
-        client: &AgentServiceClient,
+        _client: &AgentServiceClient,
         health: &HealthClient,
-        options: &mut Options,
+        _options: &mut Options,
         args: &str,
-    ) -> Result<()> {
-        super::agent_cmd_health_check(ctx, client, health, options, args)
+    ) -> (Result<()>, bool) {
+        let req: CheckRequest = match utils::make_request(args) {
+            Ok(res) => res,
+            Err(e) => {
+                return (
+                    Err(anyhow!("{:?}", e).context(REQUEST_BUILD_FAIL_MESSAGE)),
+                    false,
+                )
+            }
+        };
```

Comment:
lots of duplicated codes found in the commits. could you refactor it ?

Reply:
Update!
Were you only talking about the `make_request()` function call?

Reply:
No, IMO, maybe something like `utils::make_request(args).context(REQUEST_BUILD_FAIL_MESSAGE)?` is enough.